clients/drcachesim/reader/file_reader.h

```diff
@@ -121,10 +123,8 @@ template <typename T> class file_reader_t : public reader_t {
             return reader_t::skip_instructions(instruction_count);
         }
         // If the user asks to skip from the very start, we still need to find the chunk
-        // count marker and drain the header queue.
-        // TODO i#5538: Record all of the header values until the first timestamp
-        // and present them as new memtrace_stream_t interfaces.
-        while (chunk_instr_count_ == 0) {
+        // count marker and drain the header queue and populate the stream header values.
+        while (chunk_instr_count_ == 0 || page_size_ == 0) {
```

Comment:
Why do we need to have two conditions here? Either one of the page size or chunk instr marker should be the last one always right?

Reply:
Yes just page size; removed the chunk.  It would be nicer to wait for the timestamp but then we've already read that record and it complicates things so unfortunately we have an assumption on the final marker.  Added a comment.